ghcide/test/exe/Main.hs

```diff
@@ -4230,6 +4231,52 @@ addSigLensesTests =
             ]
         ]
 
+addSigLensesForWhereClausesTests :: TestTree
+addSigLensesForWhereClausesTests = testGroup
+  "add signature for where clauses"
+  [ testSession "Disbled" $ do
+      let content = T.unlines
+            [ "module Sigs where"
+            , "f :: b"
+            , "f = undefined"
+            , "  where"
+            , "    g = True"
+            ]
+      sendNotification SWorkspaceDidChangeConfiguration
+        $ DidChangeConfigurationParams
+        $ A.object
+            ["haskell" A..= A.object
+                ["plugin" A..= A.object
+                    ["ghcide-type-lenses" A..= A.object
+                        ["config" A..= A.object
+                            ["whereLensOn" A..= A.Bool False]]]]]
+      doc <- createDoc "Sigs.hs" "haskell" content
+      waitForProgressDone
+      lenses <- getCodeLenses doc
+      liftIO $ length lenses @?= 0
+  , test "Simple" "    g = True" "    g :: Bool\n    g = True"
+  , test "Tuple" "    (g,h) = (id, True)" "    g :: a -> a\n    (g,h) = (id, True)"
+  , test "Operator" "    g = ($)" "    g :: (a -> b) -> a -> b\n    g = ($)"
+  , test "Infix" "    a `g` b = a" "    g :: p1 -> p -> p1\n    a `g` b = a"
+  , expectFail $ test "Typeclass" "    g a b = a + b" "    g :: Num a :: a -> a -> a\n    g a b = a + b"
```

Comment:
typeclass constraints in local bindings are not displayed as top level bindings..

Reply:
There's also an ugly question about `ScopedTypeVariables`. What do we do if the type references variables bound in the top level type signature? Worth a test at least.